tensorflow_addons/seq2seq/beam_search_decoder.py

```diff
@@ -180,10 +180,12 @@ def _check_static_batch_beam_maybe(shape, batch_size, beam_width):
     """Raises an exception if dimensions are known statically and can not be
     reshaped to [batch_size, beam_size, -1]."""
     reshaped_shape = tf.TensorShape([batch_size, beam_width, None])
-    if (batch_size is not None and shape.dims[0].value is not None
-            and (shape[0] != batch_size * beam_width or
-                 (shape.ndims >= 2 and shape.dims[1].value is not None and
-                  (shape[0] != batch_size or shape[1] != beam_width)))):
+    assert len(shape.dims) > 0
+    known_size = batch_size is not None and shape.dims[0].value is not None
+    case_1 = (shape[0] == batch_size * beam_width)
+    case_2_guard = shape.ndims >= 2 and shape.dims[1].value is not None
```

Comment:
At this point, the code is quite hard to understand now. I think the more readable way is to exit early, like:

```
if not batch_size or not beam_width or not shape[0]:
  return  # Shape not static, do not check.
if shape[0] == batch_size * beam_width:
  return  
if shape.ndims >= 2 and shape[0] == batch_size and (
        shape[1] == beam_width or shape[1] is None):
     return
tf.get_logger().warn(....)
```

Reply:
That is much prettier, yes. I guess in Python having many ifs vs. one big conditional doesn't matter anyway 🙃 I'll update this PR today or tomorrow. Thanks for the feedback!